api/end_device.proto

```diff
@@ -167,6 +167,39 @@ message EndDeviceVersion {
   MessagePayloadFormatters default_formatters = 13 [(validate.rules).message.required = true];
 }
 
+// Adaptive Data Rate settings.
+message ADRSettings {
+  // Configuration options for static ADR.
+  message StaticMode {
+    // Data rate index to use.
+    DataRateIndex data_rate_index = 1 [(validate.rules).enum.defined_only = true];
+    // Transmission power index to use.
+    uint32 tx_power_index = 2 [(validate.rules).uint32.lte = 15];
+    // Number of retransmissions.
+    uint32 nb_trans = 3 [(validate.rules).uint32.lte = 15];
+  }
+
+  // Configuration options for dynamic ADR.
+  message DynamicMode {
+    // The ADR margin tells the network server how much margin it should add in ADR requests.
+    // A bigger margin is less efficient, but gives a better chance of successful reception.
+    google.protobuf.FloatValue margin = 1;
+    // Maximum number of retransmissions.
+    google.protobuf.UInt32Value max_nb_trans = 2 [(validate.rules).uint32 = {gte: 1, lte: 3}];
```

Comment:
1. Shouldn't we have both a `min_nb_trans` and `max_nb_trans`?
2. Should we then also have `{min,max}_{data_rate_index,tx_power_index}`?
3. Maybe add that if this is not set explicitly, the network default is used.